### DISCUSSION TITLE
PD info menu, PD power limit

### DISCUSSION THREAD

**Discussion by DjordjeMandic** [Ralim/IronOS#1020] [Ideas]
Any plans on usb pd info menu?

If the power is set for example to 30W but PD source can supply only 20W (based on info provided from adapter via CC lines) does the firmware limit the power to 20W or what adapter can support?

I know that there is no option to know how much power source can deliver with QC but we can know it with PD, im just qurious if current firmware uses that info to limit the power if set power in settings is above of what PD supply can deliver?

**Comment by Ralim** [comment]
> Any plans on usb pd info menu?

Yes and there is an issue to track it when its done

> 
> If the power is set for example to 30W but PD source can supply only 20W (based on info provided from adapter via CC lines) does the firmware limit the power to 20W or what adapter can support?

So the unit will pick the highest voltage it can until it exceeds the power adapters rating.
This will already bring you into a very close alignment.

So if you connect to a 35W adapter, the iron will not pick 20V. As the charger simply cant sustain the peak current draw.

In addition it does also set a secondary power limit to match the charger, so it will enforce a limit of min(user set wattage, charger reported wattage). Though generally this has zero effect as the voltage that was selected earlier will have been picked to ensure the current peak is within the chargers capabilities.
